content/docs/rfcs/13/README.md

```diff
@@ -59,139 +59,260 @@ The following are not considered as part of the adversarial model:
-  the `pageSize` amount of `WakuMessage` whose `Index` values are larger than the given `cursor`
-  (and vise versa for the backward pagination). 
-  Note that the `cursor` of a `HistoryQuery` MAY be empty (e.g., for the initial query), as such, and
-  depending on whether the  `direction` is `BACKWARD` or `FORWARD`  the last or the first `pageSize` `WakuMessage` SHALL be returned, respectively.
-
-### Sorting Messages
-The queried node MUST sort the `WakuMessage` based on their `Index`, 
-where the `senderTime` constitutes the most significant part and the `digest` comes next, and 
-then perform pagination on the sorted result. 
-As such, the retrieved page contains an ordered list of `WakuMessage` from the oldest messages to the most recent one.
-Alternatively, the `receiverTime` (instead of `senderTime` ) MAY be used to sort messages during the paging process. 
-However, it is RECOMMENDED the use of the `senderTime` for sorting as it is invariant and 
-consistent across all the nodes.
-This has the benefit of `cursor` reusability i.e., 
-a `cursor` obtained from one node can be consistently used to query from another node.
-However, this `cursor` reusability does not hold when the `receiverTime` is utilized as the receiver time is affected by the network delay and 
-nodes' clock asynchrony.
-
-### HistoryResponse
-
-RPC call to respond to a HistoryQuery call.
-- The `messages` field MUST contain the messages found,
-these are [14/WAKU2-MESSAGE](/spec/14) types.
-- `PagingInfo`  holds the paging information based on which the querying node can resume its further history queries. 
-  The `pageSize` indicates the number of returned Waku messages (i.e., the number of messages included in the `messages` field of `HistoryResponse`). 
-  The `direction` is the same direction as in the corresponding `HistoryQuery`. 
-  In the forward pagination, the `cursor` holds the `Index` of the last message in the `HistoryResponse` `messages` (and the first message in the backward paging). 
-  Regardless of the paging direction, the retrieved `messages` are always sorted in ascending order based on their timestamp as explained in the [sorting messages](#sorting-messages) section, that is, from the oldest to the most recent.
-  The requester SHALL embed the returned `cursor` inside its next `HistoryQuery` to retrieve the next page of the [14/WAKU2-MESSAGE](/spec/14).  
-  The `cursor` obtained from one node SHOULD NOT be used in a request to another node because the result may be different.
-- The `error` field contains information about any error that has occurred while processing the corresponding `HistoryQuery`.
-  `NONE` stands for no error.
-  This is also the default value.
-  `INVALID_CURSOR` means that the `cursor` field of `HistoryQuery` does not match with the `Index` of any of the `WakuMessage` persisted by the queried node.
+### Waku message sorting
+
+The key-value entries in the store MUST be time-sorted by the `WakuMessage` `timestamp` attribute.
```

Comment:
Would't it better to also indicate that messages can be stored grouping by pubsubTopic and in fact segregated by contentTopic for easy management and search?

I am thinking it would narrow search space this way and also help to come up with different architectures of underlying Stores.

Reply:
If I understand your suggestion correctly, this is about the actual storage schema and improving that? This would be the domain of the archive, which this RFC is not concerned with. The aim is simply to design the query interface language with no recommendations on how persistence/retrieval can be optimised.

Reply:
Ok, but i am worried about a query being done on whole archive all the time whereas almost all the times the client querying is aware of pubsub and contentTopics for which it is querying. 
Was not aware that archive was a different rfc, will go through it.

Reply:
True, but this would still be up to how the Store service node decides to efficiently store and query nodes in the DB and should not be included in specification for the query language. There is no RFC for the archive currently.